toolchains/jdk_build_file.bzl

```diff
@@ -83,6 +83,9 @@ java_runtime(
         ":jdk-lib",
         ":jre",
     ],
+    # Provide the 'java` binary explicitly so that the correct path is used by
+    # Bazel even when the host platform differs from the execution platform.
+    java = glob(["bin/java.exe", "bin/java"])[0],
```

Comment:
Should we also:

1. Document that `java_runtime.java` should (must?) be explicitly set, otherwise the default will be a guess based on the host platform?
1. Update the TODO in `JavaRuntime.java` accordingly?

Reply:
(Whoops, I just noticed that this is not the main Bazel repo, but the comments still apply outside of this PR)

Reply:
> 1. Document that `java_runtime.java` should (must?) be explicitly set, otherwise the default will be a guess based on the host platform?
> 2. Update the TODO in `JavaRuntime.java` accordingly?

Documenting it as recommended sounds good to me.

I wonder if there's a path to making it mandatory and removing the logic for inferring it, but I'm not sure what the cleanup for that would look like.